packages/venia-concept/src/reducers/checkout.js

```diff
@@ -12,8 +12,11 @@ const storedShippingMethod = storage.getItem('shippingMethod');
 
 const initialState = {
     editing: null,
-    paymentMethod: storedPaymentMethod && storedPaymentMethod.code,
-    paymentTitle: storedPaymentMethod && storedPaymentMethod.title,
+    paymentCode: storedPaymentMethod && storedPaymentMethod.code,
+    paymentDisplayPrimary:
+        storedPaymentMethod && storedPaymentMethod.display_primary,
+    paymentDisplaySecondary:
+        storedPaymentMethod && storedPaymentMethod.display_secondary,
```

Comment:
`paymentCode` is a much less ambiguous name for what this value holds.

I split `paymentTitle` into `paymentDisplayPrimary` and `paymentDisplaySecondary` to facilitate the desired UI.